samples/grids/grid/state-persistence-main/src/index.ts

```diff
@@ -0,0 +1,134 @@
+        var allCheckboxes = Array.from(document.getElementsByTagName("igc-checkbox"));
+
+        this._bind = () => {
+            registerIconFromText("restore", restoreIcon, "material");
+            registerIconFromText("save", saveIcon, "material");
+            registerIconFromText("clear", clearIcon, "material");
+            registerIconFromText("forward", forwardIcon, "material");
+            registerIconFromText("delete", deleteIcon, "material");
+            registerIconFromText("refresh", refreshIcon, "material");
+
+            grid.data = this.gridData;
+            grid.allowAdvancedFiltering = true;
+            grid.filterMode = 'excelStyleFilter';
+
+            saveStateBtn.addEventListener('click', (ev: any) => { this.saveGridState(); });
+            restoreStateBtn.addEventListener('click', (ev: any) => { this.restoreGridState(); });
+            resetStateBtn.addEventListener('click', (ev: any) => { this.resetGridState(); });
+            leavePageBtn.addEventListener('click', (ev: any) => { this.leavePage(); });
+            leaveLink.addEventListener('click', (ev: any) => { this.saveGridState(); });
+            clearStorageBtn.addEventListener('click', (ev: any) => { this.clearStorage(); });
+            reloadPageBtn.addEventListener('click', (ev: any) => { this.reloadPage(); });
+    
+            allCheckboxes.forEach(cb => {
+                cb.addEventListener("igcChange", (ev: CustomEvent) => { this.onChange(ev, cb.id); });
+            });
+
+            window.addEventListener("load", () => { this.restoreGridState(); })
+            window.addEventListener("beforeunload", () => { this.saveGridState(); });
+        }
+        this._bind();
+    }
+
+    public saveGridState() {
+        const state = this.gridState.getStateAsString();
+        window.localStorage.setItem(this.stateKey, state);
+    }
+
+    public restoreGridState() {
+        const state = window.localStorage.getItem(this.stateKey);
+        if (state) {
+            this.gridState.applyStateFromString(state);
+        }
+    }
+
+    public resetGridState() {
+        const pagingState = { index: 0, recordsPerPage: 15, metadata: { countPages: 3, countRecords: this.gridData.length } };
+        (this.grid as any).pagingState = pagingState;
+        this.grid.clearFilter();
+        this.grid.sortingExpressions = [];
+        this.grid.groupingExpressions = [];
+        this.grid.deselectAllColumns();
+        this.grid.deselectAllRows();
+        this.grid.clearCellSelection();
+    }
+
+    public onChange(event: CustomEvent, action: string) {
+        if (action === 'allFeatures') {
+            var allCheckboxes = Array.from(document.getElementsByTagName("igc-checkbox"));
+            allCheckboxes.forEach(cb => {
+                cb.checked = event.detail;
+            });
+            for (const key of Object.keys(this.options)) {
+                this.gridState.options[key] = event.detail;
```

Comment:
Here it seems to throw error:
 TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'IgcGridStateOptions'.
  No index signature with a parameter of type 'string' was found on type 'IgcGridStateOptions'.

Probably needs explicit cast to any.